boards: Add the Nordic nRF5340 audio DK

koffes opened the pull request:
Migration of boardfiles from sdk-nrf to Zephyr.

fixes https://github.com/zephyrproject-rtos/zephyr/issues/47296

koffes committed:
boards: Added nRF5340 Audio DK (PCA10121)

Copied from nRF Connect SDK and edited
Co-authored-by: Grzegorz Ferenc

Signed-off-by: Kristoffer Rist Skøien <kristoffer.skoien@nordicsemi.no>

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/board.h:
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Needs to be apache

koffes replied:
Done

Thalley commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_common.dts:
```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2020-2022 Nordic Semiconductor ASA
```
How come only a subset of the files have Copyright from 20XX-2022, and some are just 2019?

koffes replied:
This is the same format as nrf5340dk_nrf5340. Though, I believe this is the old way, and now just the "first year" should be used.  For the CI building, can you comment @anangl or @nordicjm ?

Thalley replied:
Possibly we can just add `nrf5340_audio_dk_nrf5340_cpuapp` to some of of the `sample.yaml` files we have in the `platform_allow` or `integration_platforms` like what we've done with `nrf5340dk_nrf5340_cpuapp`

koffes replied:
Done

Thalley reviewed (commented):
@koffes Is the CI building anything for this board? In other words, are we in this PR testing that these board files are valid and an application can be built for the netcore and the appcore?

nordicjm reviewed (changes requested):
1) Squash into 1 commit
2) This board does not build:
```
-- Found BOARD.dts: /tmp/aa/zephyr/boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp.dts
devicetree error: 'label' is marked as deprecated in 'properties:' in /tmp/aa/zephyr/dts/bindings/sdhc/zephyr,sdhc-spi-slot.yaml for node /soc/peripheral@50000000/spi@a000/sdhc@0.
-- In: /tmp/aa/zephyr/twister-out/nrf5340_audio_dk_nrf5340_cpuapp/samples/drivers/audio/dmic/sample.drivers.audio.dmic/zephyr
```
3) Looks to be targeting an old version of zephyr, compare against the nrf5340dk board and update as needed:
```
warning: RPMSG_SERVICE_IPM_TX_NAME (defined at boards/arm/nrf5340_audio_dk_nrf5340/Kconfig.defconfig:96) defined without a type

warning: RPMSG_SERVICE_IPM_RX_NAME (defined at boards/arm/nrf5340_audio_dk_nrf5340/Kconfig.defconfig:100) defined without a type
```

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/CMakeLists.txt:
```diff
@@ -0,0 +1,8 @@
+    AND CONFIG_BOARD_ENABLE_CPUNET)
+zephyr_library()
+zephyr_library_sources(nrf5340_audio_dk_cpunet_reset.c)
+endif()
```
nrf5340dk file additionally has this:
```
if (CONFIG_BUILD_WITH_TFM)
  zephyr_library_include_directories(
    $<TARGET_PROPERTY:tfm,TFM_BINARY_DIR>/install/interface/include
  )
endif()
```

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/Kconfig.defconfig:
```diff
@@ -0,0 +1,142 @@
+config RPMSG_SERVICE_DUAL_IPM_SUPPORT
+	default RPMSG_SERVICE
+
+if RPMSG_SERVICE_DUAL_IPM_SUPPORT
+
+config IPM_MSG_CH_0_ENABLE
+	default y
+
+config IPM_MSG_CH_1_ENABLE
+	default y
+
+config RPMSG_SERVICE_IPM_TX_NAME
+	default "IPM_0" if RPMSG_SERVICE_MODE_MASTER
+	default "IPM_1" if RPMSG_SERVICE_MODE_REMOTE
+
+config RPMSG_SERVICE_IPM_RX_NAME
+	default "IPM_1" if RPMSG_SERVICE_MODE_MASTER
```
Lines are not present in nrf5340dk - are they needed?

koffes replied:
Removed now.

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/Kconfig.defconfig:
```diff
@@ -0,0 +1,142 @@
+config HEAP_MEM_POOL_SIZE
+	default 4096 if BT_RPMSG
+
+config BT_HCI_VS
+	default y if BT
```
Not present in nrf5340dk - is this needed?

koffes replied:
Removed

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/Kconfig.defconfig:
```diff
@@ -0,0 +1,142 @@
+config BT_CTLR
+	default y if BT
+
+config BT_ECC
+	default y if BT
```
Not in nrf5340dk - needed?

koffes replied:
Removed

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/Kconfig.defconfig:
```diff
@@ -0,0 +1,142 @@
+config BT_ECC
+	default y if BT
+
+endif # BOARD_NRF5340_AUDIO_DK_NRF5340_CPUNET
```
nrf5340dk has additionally:
```
config UART_NRF_DK_SERIAL_WORKAROUND
  default y if ZTEST

# When the nRF DK serial workaround is enabled, increase the stack size in
# tests built with no optimizations, as the standard size may be insufficient
# then. Use 512 to align with the FPU_SHARING case, where the default value
# is also modified.
config TEST_EXTRA_STACK_SIZE
  default 512 if UART_NRF_DK_SERIAL_WORKAROUND && NO_OPTIMIZATIONS
```

koffes replied:
Added

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/board.cmake:
```diff
@@ -0,0 +1,16 @@
+if(CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP OR CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP_NS)
+board_runner_args(jlink "--device=nrf5340_xxaa_app" "--speed=4000")
+endif()
+
```
nrf5340dk additionally has:
```
if(CONFIG_TFM_FLASH_MERGED_BINARY)
  set_property(TARGET runners_yaml_props_target PROPERTY hex_file "${CMAKE_BINARY_DIR}/tfm_merged.hex")
endif()
```

koffes replied:
Done

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp.yaml:
```diff
@@ -0,0 +1,19 @@
+  - i2c
+  - pwm
+  - watchdog
+  - usb_cdc
```
Don't think usb_cdc is, and not sure if there is one for audio

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_common.dts:
```diff
@@ -0,0 +1,162 @@
+		compatible = "zephyr,sdhc-spi-slot";
+		reg = <0>;
+		status = "okay";
+		label = "SDHC0";
```
No labels

koffes replied:
I have removed the labels in this file. We shall still keep labels elsewhere? I see that there are quite a few labels on other boards.

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_common.dts:
```diff
@@ -0,0 +1,162 @@
+	};
+	cs47l63: cs47l63@1 {
+		reg = <1>;
+		label = "CS47L63";
```
No labels

koffes replied:
Done

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_common.dts:
```diff
@@ -0,0 +1,162 @@
+		slot1_ns_partition: partition@c0000 {
+			label = "image-1-nonsecure";
+		};
+		scratch_partition: partition@f0000 {
+			label = "image-scratch";
+			reg = <0x000f0000 0xa000>;
+		};
```
Scratch not needed unless waiting to use mcuboot swap with scratch which is not recommended

koffes replied:
@pirun could you comment here ref. the DFU solution we have today?

pirun replied:
We don't use scratch swap on LE Audio project.
This is safe to remove.

koffes replied:
Removed

koffes replied:
Removed

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_defconfig:
```diff
@@ -0,0 +1,60 @@
+CONFIG_SOC_NRF5340_CPUAPP_QKAA=y
+CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP=y
+
+CONFIG_BOARD_ENABLE_DCDC_APP=y
+CONFIG_BOARD_ENABLE_DCDC_NET=y
+CONFIG_BOARD_ENABLE_CPUNET=y
```
Not sure that these should be here

koffes replied:
Removed.

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_defconfig:
```diff
@@ -0,0 +1,60 @@
+CONFIG_GPIO=y
+CONFIG_NFCT_PINS_AS_GPIOS=y
+
+CONFIG_BT=y
```
The basic default for a board needs to have all peripherals disabled and only allows for UART and timer to be enabled

koffes commented:
Thanks @nordicjm. I see that there are further changes which has been introduced to the boardfiles. I am going through the files one by one, comparing to nRF5340 DK. Will update this PR soon.

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_defconfig:
```diff
@@ -0,0 +1,56 @@
+
+CONFIG_BT=y
+
+# Custom solution used instead of Zephyr I2S driver
+CONFIG_I2S=n
+CONFIG_NRFX_I2S=y
+
+CONFIG_I2C=y
+CONFIG_SPI=y
+CONFIG_ADC=y
+
+# Enable NRFX_CLOCK for ACLK control
+CONFIG_NRFX_CLOCK=y
+
+# Minimal libc is not used. Note that this will impact error codes
+CONFIG_NEWLIB_LIBC=y
+
+# Audio codec LC3 related defines
+# FPU_SHARING enables preservation of the hardware floating point registers
+# across context switches to allow multiple threads to perform concurrent
+# floating point operations.
+CONFIG_FPU=y
+CONFIG_FPU_SHARING=y
+
+# Enable SDHC interface
+CONFIG_DISK_DRIVERS=y
+CONFIG_DISK_DRIVER_SDMMC=y
+
+# Allocate RAM buffer for transferring chunck of data from Flash to SPI
+CONFIG_SPI_NRFX_RAM_BUFFER_SIZE=8
+
+# Config the file system
+CONFIG_FILE_SYSTEM=y
+CONFIG_FAT_FILESYSTEM_ELM=y
+CONFIG_FS_FATFS_LFN=y
+# Using stack for LFN work queue
+CONFIG_FS_FATFS_LFN_MODE_STACK=y
+# exFAT enabled to support longer file names and higher transfer speed
+CONFIG_FS_FATFS_EXFAT=y
+CONFIG_FS_FATFS_MAX_LFN=255
```
This should not be in the defconfig file. The board by default should be barebones. I cloned this and started running the tests and they all failed, reason being because every test is trying to pull in the FAT filesystem module, which is a module I excluded.

koffes replied:
Done

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpunet_defconfig:
```diff
@@ -0,0 +1,22 @@
+CONFIG_GPIO=y
+
+
+CONFIG_HEAP_MEM_POOL_SIZE=8192
```
Same with this

koffes replied:
Done

carlescufi commented:
@anangl @gmarull please take a look (not urgent)

nordicjm commented:
Watchdog does not seem to be configured correctly:
```
/tmp/aa/zephyr/samples/drivers/watchdog/src/main.c: In function 'main':
/tmp/aa/zephyr/include/zephyr/device.h:104:39: error: '__device_dts_ord_DT_N_ALIAS_watchdog0_ORD' undeclared (first use in this function)
  104 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                                       ^~~~~~~~~
/tmp/aa/zephyr/include/zephyr/toolchain/common.h:132:26: note: in definition of macro '_DO_CONCAT'
  132 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/tmp/aa/zephyr/include/zephyr/device.h:104:31: note: in expansion of macro '_CONCAT'
  104 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                               ^~~~~~~
/tmp/aa/zephyr/include/zephyr/device.h:274:37: note: in expansion of macro 'DEVICE_NAME_GET'
  274 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_NAME(node_id))
      |                                     ^~~~~~~~~~~~~~~
/tmp/aa/zephyr/include/zephyr/device.h:292:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  292 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/tmp/aa/zephyr/samples/drivers/watchdog/src/main.c:61:42: note: in expansion of macro 'DEVICE_DT_GET'
   61 |         const struct device *const wdt = DEVICE_DT_GET(DT_ALIAS(watchdog0));
      |                                          ^~~~~~~~~~~~~
/tmp/aa/zephyr/include/zephyr/device.h:104:39: note: each undeclared identifier is reported only once for each function it appears in
  104 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                                       ^~~~~~~~~
/tmp/aa/zephyr/include/zephyr/toolchain/common.h:132:26: note: in definition of macro '_DO_CONCAT'
  132 | #define _DO_CONCAT(x, y) x ## y
      |                          ^
/tmp/aa/zephyr/include/zephyr/device.h:104:31: note: in expansion of macro '_CONCAT'
  104 | #define DEVICE_NAME_GET(name) _CONCAT(__device_, name)
      |                               ^~~~~~~
/tmp/aa/zephyr/include/zephyr/device.h:274:37: note: in expansion of macro 'DEVICE_NAME_GET'
  274 | #define DEVICE_DT_NAME_GET(node_id) DEVICE_NAME_GET(Z_DEVICE_DT_DEV_NAME(node_id))
      |                                     ^~~~~~~~~~~~~~~
/tmp/aa/zephyr/include/zephyr/device.h:292:34: note: in expansion of macro 'DEVICE_DT_NAME_GET'
  292 | #define DEVICE_DT_GET(node_id) (&DEVICE_DT_NAME_GET(node_id))
      |                                  ^~~~~~~~~~~~~~~~~~
/tmp/aa/zephyr/samples/drivers/watchdog/src/main.c:61:42: note: in expansion of macro 'DEVICE_DT_GET'
   61 |         const struct device *const wdt = DEVICE_DT_GET(DT_ALIAS(watchdog0));
      |                                          ^~~~~~~~~~~~~
```

nordicjm commented:
Possible i2s configuration issue:
```
*** Booting Zephyr OS build zephyr-v3.1.0-5046-geb8b98bfd02b  ***
Running TESTSUITE i2s_speed_test
===================================================================
START - test_i2s_tx_transfer_configure
Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c:187: test_i2s_tx_transfer_configure: (dev_i2s_tx is NULL)
device I2S_0 not found
FAIL - test_i2s_tx_transfer_configure in 0.030 seconds
===================================================================
START - test_i2s_rx_transfer_configure
Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/i2s/i2s_speed/src/test_i2s_speed.c:199: test_i2s_rx_transfer_configure: (dev_i2s_rx is NULL)
device I2S_0 not found
FAIL - test_i2s_rx_transfer_configure in 0.036 seconds
```

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts:
```diff
@@ -0,0 +1,133 @@
+
+		curr_mon_alert_in: curr_mon_alert_in {
+			gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
+			label = "Curr mon alert in from INA231";
```
this looks wrong, why a driver specific GPIO under `gpio-keys`?

koffes replied:
We have a task to correct this in NCS, but we need that resolved first before this can be removed.

gmarull replied:
this cannot get in as-is.

koffes replied:
We are working on changing out the entire INA231 driver to the upstream one, but this will take some time and it would have been great to get this merged despite of the ongoing efforts on that front. Do you have any suggestions for a temporary solution?

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts:
```diff
@@ -0,0 +1,133 @@
+
+		hw_codec_sel_out: hw_codec_sel_out {
+			gpios = <&gpio0 21 GPIO_ACTIVE_LOW>;
+			label = "HW codec selector. Low selects OB HW codec, high \
+				 selects external HW codec";
```
label misuse, it must not be used for documentation (check others as well)

koffes replied:
Altered

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts:
```diff
@@ -0,0 +1,133 @@
+		compatible = "gpio-leds";
+		rgb1_red: led_0 {
+			gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
+			label = "0 LED_RGB_RED CORE_APP";
+		};
+		rgb1_green: led_1 {
+			gpios = <&gpio0 25 GPIO_ACTIVE_HIGH>;
+			label = "0 LED_RGB_GREEN CORE_APP";
+		};
+		rgb1_blue: led_2 {
+			gpios = <&gpio0 26 GPIO_ACTIVE_HIGH>;
+			label = "0 LED_RGB_BLUE CORE_APP";
+		};
+		rgb2_red: led_3 {
+			gpios = <&gpio0 28 GPIO_ACTIVE_HIGH>;
+			label = "1 LED_RGB_RED CORE_NET";
+		};
+		rgb2_green: led_4 {
+			gpios = <&gpio0 29 GPIO_ACTIVE_HIGH>;
+			label = "1 LED_RGB_GREEN CORE_NET";
+		};
+		rgb2_blue: led_5 {
+			gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+			label = "1 LED_RGB_BLUE CORE_NET";
+		};
+		led1_blue: led_6 {
+			gpios = <&gpio0 31 GPIO_ACTIVE_HIGH>;
+			label = "2 LED_MONO_BLUE CORE_APP";
+		};
+		led2_green: led_7 {
+			gpios = <&gpio1 0 GPIO_ACTIVE_HIGH>;
+			label = "3 LED_MONO_GREEN CORE_APP";
+		};
+		led3_green: led_8 {
+			gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
+			label = "4 LED_MONO_GREEN CORE_APP";
```
-1 on encoding "structured" information on labels

koffes replied:
Please have a look now and see what you think.

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpunet_defconfig:
```diff
@@ -0,0 +1,19 @@
+# enable GPIO
+CONFIG_GPIO=y
+
+CONFIG_BT=y
```
BT is not minimal, needs to be enabled by apps requiring it

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,145 @@
+	ARG_UNUSED(dev);
+
+	ret = core_config();
+	if (ret) {
```
`ret < 0`

koffes replied:
This can be changed, but this function is supposed to only return 0.

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,145 @@
+
+	if (!device_is_ready(board_id_en.port)) {
+		LOG_ERR("GPIO is not ready!");
+		return -ENXIO;
```
`-ENODEV`

koffes replied:
Changed

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,145 @@
+#define DEBUG_SETUP()
+#endif
+
+#define SHARED_SPI DT_NODELABEL(spi4) /* SD card and HW codec share the SPI4 */
```
unused?

koffes replied:
Removed

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,145 @@
+LOG_MODULE_REGISTER(nrf5340_audio_dk_nrf5340_cpuapp, CONFIG_LOG_DEFAULT_LEVEL);
+
+#if defined(CONFIG_BT_CTLR_DEBUG_PINS_CPUAPP)
+#include <../subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/debug.h>
```
why relative include here?

koffes replied:
This is copied from the ordinary nrf5340dk_nrf5340

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,145 @@
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+
+#include <soc.h>
```
why is this needed?

koffes replied:
This was added to the standard nrf5340dk_nrf5340 in commit 86a1cc7 to fix a possible compilation error.

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,145 @@
+#include <soc.h>
+#include <zephyr/drivers/gpio.h>
+
+#include "nrfx_gpiote.h"
```
this is not a local include, use `<>`

koffes replied:
Changed

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,145 @@
+	ret = gpio_pin_configure_dt(&pmic_iset, GPIO_OUTPUT_LOW);
+	if (ret) {
+		return ret;
+	}
+
+	/* Set on-board DSP/HW codec as default */
+	static const struct gpio_dt_spec hw_codec_sel =
+		GPIO_DT_SPEC_GET(DT_NODELABEL(hw_codec_sel_out), gpios);
+
+	if (!device_is_ready(hw_codec_sel.port)) {
+		LOG_ERR("GPIO is not ready!");
+		return -ENXIO;
+	}
+
+	ret = gpio_pin_configure_dt(&hw_codec_sel, GPIO_OUTPUT_LOW);
+	if (ret) {
+		return ret;
+	}
+
+	/* Pull the CS47L63 reset line to high (this pin is active low) */
+	static const struct gpio_dt_spec hw_codec_reset =
+		GPIO_DT_SPEC_GET(DT_NODELABEL(hw_codec_reset_out), gpios);
+
+	if (!device_is_ready(hw_codec_reset.port)) {
+		LOG_ERR("GPIO is not ready!");
+		return -ENXIO;
+	}
+
+	ret = gpio_pin_configure_dt(&hw_codec_reset, GPIO_OUTPUT_HIGH);
+	if (ret) {
+		return ret;
+	}
+
+	/* Disable board revision readback as default */
+	static const struct gpio_dt_spec board_id_en =
+		GPIO_DT_SPEC_GET(DT_NODELABEL(board_id_en_out), gpios);
+
+	if (!device_is_ready(board_id_en.port)) {
+		LOG_ERR("GPIO is not ready!");
+		return -ENXIO;
```
many of these init configs deserve better explanation, and possibly proper description in Devicetree.

koffes replied:
Have changed the descriptions and added comments.

larsgk commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts:
```diff
@@ -0,0 +1,133 @@
+/ {
+leds:	leds {
+		compatible = "gpio-leds";
```
Does the main RGB LED (center) have PWM capability? If so, it would be nice to add the pwmleds & aliases.

koffes replied:
We can do that, but I would like to keep it separate from this PR. In the current stage, we want to move this board and not add or remove any features.

joerchan commented on modules/trusted-firmware-m/Kconfig.tfm:
```diff
@@ -11,6 +11,7 @@ config TFM_BOARD
 	string
 	default "nordic_nrf/nrf9160dk_nrf9160" if BOARD_NRF9160DK_NRF9160_NS
 	default "nordic_nrf/nrf5340dk_nrf5340_cpuapp" if BOARD_NRF5340DK_NRF5340_CPUAPP_NS
+	default "nordic_nrf/nrf5340_audio_dk_nrf5340_cpuapp" if BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP_NS
```
This is not a valid TF-M build target.

Remove this.

joerchan commented on boards/arm/nrf5340_audio_dk_nrf5340/Kconfig.defconfig:
```diff
@@ -0,0 +1,106 @@
+config BOARD
+	default "nrf5340_audio_nrf5340_cpuapp" if BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP || BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP_NS
+
+if BUILD_WITH_TFM
+
+# By default, if we build with TF-M, instruct build system to
+# flash the combined TF-M (Secure) & Zephyr (Non Secure) image
+config TFM_FLASH_MERGED_BINARY
+	bool
+	default y
+
+endif # BUILD_WITH_TFM
```
I think you should remove this part as well. This keeps it consistent with the other nrf53 boards.

koffes replied:
Removed

anangl commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp.dts:
```diff
@@ -0,0 +1,38 @@
+	};
+};
+
+zephyr_udc0: &usbd {
+	compatible = "nordic,nrf-usbd";
+	status = "okay";
+
+	hs_0: hs_0 {
+		compatible = "usb-audio-hs";
+		mic-feature-mute;
+		mic-channel-l;
+		mic-channel-r;
+
+		hp-feature-mute;
+		hp-channel-l;
+		hp-channel-r;
+	};
+};
```
Why is it here and in `nrf5340_audio_dk_nrf5340_cpuapp_ns.dts` instead of in `nrf5340_audio_dk_nrf5340_cpuapp_common.dts`?

anangl commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpunet_defconfig:
```diff
@@ -0,0 +1,28 @@
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+
+CONFIG_PINCTRL=y
```
It's already enabled in [line 14](https://github.com/zephyrproject-rtos/zephyr/pull/49720/files#diff-3f53920eb6b0749f2393c5f28712dbc30c2b317159d020ba472b3fc007ebe5c7R14).

koffes replied:
Thanks, removed.

joerchan commented on boards/arm/nrf5340_audio_dk_nrf5340/Kconfig:
```diff
@@ -0,0 +1,56 @@
+
+config BOARD_ENABLE_CPUNET
+	bool "nRF53 Network MCU"
+	help
```
Please add:
```
	select SOC_NRF_GPIO_FORWARDER_FOR_NRF5340 if \
		$(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF_GPIO_FORWARDER))
```

See: https://github.com/zephyrproject-rtos/zephyr/pull/50556

koffes replied:
Done

Thalley commented on samples/bluetooth/iso_broadcast_benchmark/sample.yaml:
```diff
@@ -4,5 +4,5 @@ sample:
   sample.bluetooth.iso_broadcast_benchmark:
     build_only: true
-    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp
+    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf5340_audio_dk_nrf5340_cpuapp
```
We should also add CI builds for the `nrf5340_audio_dk_nrf5340_cpunet`

Thalley replied:
Generally in samples where we already have `nrf5340dk_nrf5340_cpunet`, so something like `hci_rpmsg` and the `hci_uart` would make sense I think.

We may also want to copy the `.conf` files used for the nRF5340DK for those samples. @cvinayak what do you think?

cvinayak replied:
For CI builds, add the board under `integration_platforms` of the `hci_rpmsg` sample's `sample.yaml` for ISO Broadcast and ISO Receive configurations, here:

https://github.com/zephyrproject-rtos/zephyr/blob/ba810303c5cf489982b0c034d9db848217aaf7ea/samples/bluetooth/hci_rpmsg/sample.yaml#L12-L25

koffes replied:
Done

greg-fer reviewed (changes requested):
We need a doc page for this addition, similar to https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/boards/arm/nrf52840dk_nrf52840/doc/index.html 

From what I can see in the board dir history, whenever a board is added to the repo, the docs are there from the start.

tejlmand reviewed (changes requested):
Please follow the board contribution guide:
https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#contributing-your-board

some parts are missing in this PR.

Thalley commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,165 @@
+.. _thingy53_nrf5340:
```
Not a thingy53 :)

koffes replied:
Quite picky of you, but OK, I will fix that.

Thalley commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,165 @@
+********
+
+The nRF5340 Audio DK (PCA10121) is designed for showcasing, developing and experimenting
+with Bluetooth LE Audio. Keywords: Auracast™, broadcast, BIS, unicast, CIS, headsets, microphones.
```
Seems redundant both write broadcast and BIS, as well as unicast and CIS. They are, in this context, the same, right?

greg-fer replied:
Addressed in my suggestion.

koffes replied:
That is true. Added this to match on keyword search.

Thalley commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,165 @@
+============
+* A full-featured Arm Cortex-M33F core with DSP instructions,
+  FPU, and Armv8-M Security Extension, running at up to 128 MHz, 
+  referred to as the **application core**
```
```suggestion
  referred to as the **application core**.
```

Thalley commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,165 @@
+   The nRF5340 has a flash read-back protection feature. When flash read-back
+   protection is active, you will need to recover the chip before reflashing.
+   If you are flashing with :ref:`west <west-build-flash-debug>`, run
+   this command for more details on the related ``--recover`` option:
+
+   .. code-block:: console
+
+      west flash -H -r nrfjprog --skip-rebuild
```
Am I missing something, or how is the command here related to the `--recover` option?

koffes replied:
It provides information on the --recover option.

Thalley replied:
I see :)

greg-fer commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,165 @@
+
+.. figure:: img/nrf5340_audio_dk.jpg
+     :align: center
+     :alt: nRF5340 DK
```
```suggestion
     :alt: nRF5340 Audio DK
```

greg-fer commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,165 @@
+Overview
+********
+
+The nRF5340 Audio DK (PCA10121) is designed for showcasing, developing and experimenting
+with Bluetooth LE Audio. Keywords: Auracast™, broadcast, BIS, unicast, CIS, headsets, microphones.
+Zephyr uses the nrf5340_audio_dk_nrf5340 board configuration for building
+for the nRF5340 Audio DK. 
```
```suggestion
The nRF5340 Audio DK (PCA10121) is designed for showcasing, developing and experimenting
with Bluetooth® LE Audio.
Zephyr uses the nrf5340_audio_dk_nrf5340 board configuration for building
for the nRF5340 Audio DK.

You can use this board for developing LE-Audio-compatible applications that support Auracast™, both the connected isochronous stream (CIS) mode and the broadcast isochronous stream (BIS) modes,  broadcast and unicast, and offer support for both headsets and microphones.
```

koffes commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,165 @@
+| WDT       | on-chip    | watchdog             |
++-----------+------------+----------------------+
+
+Other hardware features have not been enabled yet for this board.
+See `Nordic Semiconductor Infocenter`_
+for a complete list of nRF5340 Audio DK board hardware features.
```
That is currently correct, but the nRF5340 is there. I will alter this section and push a fix today.

larsgk commented:
I tried the PR against a sample I did using zephyr i2s with a generated sine wave and found I had to put the following in the local proj files to make it not assert:

in proj ( to avoid ASSERTION FAIL [m_cb.state != NRFX_DRV_STATE_UNINITIALIZED] ):
```
CONFIG_NEWLIB_LIBC=y
CONFIG_FPU=y
CONFIG_FPU_SHARING=y
```

If these are left out of the default config in the PR on purpose (and meant to be set in the app), the PR looks good to me (r+)

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/Kconfig.defconfig:
```diff
@@ -0,0 +1,96 @@
+
+config BOARD
+	default "nrf5340_audio_dk_nrf5340_cpuapp" if BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP || BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP_NS
+
```
Normal nrf5340dk board has this here:
```
# By default, if we build for a Non-Secure version of the board,
# enable building with TF-M as the Secure Execution Environment.
config BUILD_WITH_TFM
        default y if BOARD_NRF5340DK_NRF5340_CPUAPP_NS

if BUILD_WITH_TFM

# By default, if we build with TF-M, instruct build system to
# flash the combined TF-M (Secure) & Zephyr (Non Secure) image
config TFM_FLASH_MERGED_BINARY
        bool
        default y

endif # BUILD_WITH_TFM
```

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,103 @@
+Zephyr uses the nrf5340_audio_dk_nrf5340 board configuration for building
+for the nRF5340 Audio DK.
+
+You can use this board for developing LE-Audio-compatible applications that support Auracast™, both the connected isochronous stream (CIS) mode and the broadcast isochronous stream (BIS) modes,  broadcast and unicast, and offer support for both headsets and microphones.
```
Cut this line into multiple

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,103 @@
+     :alt: nRF5340 DK
+
+More information about the board can be found at the `nRF5340 Audio DK website`_.
+nRF5340 SoC
```
Add newline before new section

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,103 @@
+More information about the board can be found at the `nRF5340 Audio DK website`_.
+nRF5340 SoC
+===========
+The nRF5340 Audio DK is built around the nRF5340 SoC, which has the following characteristics:
```
And newline after header

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_defconfig:
```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2020 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_NRF53X=y
+CONFIG_SOC_NRF5340_CPUAPP_QKAA=y
+CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP=y
+
+CONFIG_ARM_MPU=y
+
+CONFIG_HW_STACK_PROTECTION=y
+
+CONFIG_ARM_TRUSTZONE_M=y
+
+CONFIG_GPIO=y
+
+CONFIG_SERIAL=y
+
```
Comments from original file seem to have been removed?

koffes replied:
Yes. I can add them back if required.
I dont feel comments on this format
`# Enable hardware stack protection
CONFIG_HW_STACK_PROTECTION=y

# Enable TrustZone-M
CONFIG_ARM_TRUSTZONE_M=y

# enable GPIO
CONFIG_GPIO=y`

Adds value. These configs are easily searchable in the Zephyr docs.

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpunet.dts:
```diff
@@ -0,0 +1,93 @@
+			label = "image-1";
+			reg = <0x0001E000 0x12000>;
+		};
+		scratch_partition: partition@30000 {
+			label = "image-scratch";
+			reg = <0x00030000 0xa000>;
+		};
```
This can go unless the netcore update is using this

koffes replied:
@pirun I believe you stated earlier that the scratch was not required?

nordicjm replied:
Unresolved

koffes commented:
> I tried the PR against a sample I did using zephyr i2s with a generated sine wave and found I had to put the following in the local proj files to make it not assert:
> 
> in proj ( to avoid ASSERTION FAIL [m_cb.state != NRFX_DRV_STATE_UNINITIALIZED] ):
> 
> ```
> CONFIG_NEWLIB_LIBC=y
> CONFIG_FPU=y
> CONFIG_FPU_SHARING=y
> ```
> 
> If these are left out of the default config in the PR on purpose (and meant to be set in the app), the PR looks good to me (r+)

They are left out on purpose, but I find it strange that omitting these would cause an assert.

larsgk commented:
> > I tried the PR against a sample I did using zephyr i2s with a generated sine wave and found I had to put the following in the local proj files to make it not assert:
> > in proj ( to avoid ASSERTION FAIL [m_cb.state != NRFX_DRV_STATE_UNINITIALIZED] ):
> > ```
> > CONFIG_NEWLIB_LIBC=y
> > CONFIG_FPU=y
> > CONFIG_FPU_SHARING=y
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If these are left out of the default config in the PR on purpose (and meant to be set in the app), the PR looks good to me (r+)
> 
> They are left out on purpose, but I find it strange that omitting these would cause an assert.

I am using cmsis/arm math to generate a sine wave while in a busy thread sending data over i2s.  It's probably ok to leave it out as it might be a special case.

greg-fer reviewed (changes requested):
Just one minor link issue left in the docs.

Thalley commented on boards/arm/nrf5340_audio_dk_nrf5340/CMakeLists.txt:
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA.
```
Minor, but you could consider adding the current year for the copyright notices in these files, e.g. as
```suggestion
# Copyright (c) 2021-2022 Nordic Semiconductor ASA.
```

But that's optional

Thalley reviewed (dismissed):
LGTM

cvinayak commented on samples/bluetooth/hci_rpmsg/sample.yaml:
```diff
@@ -16,6 +16,7 @@ tests:
     tags: bluetooth
     integration_platforms:
       - nrf5340dk_nrf5340_cpunet
+      - nrf5340_audio_dk_nrf5340_cpunet
```
Add `nrf5340_audio_dk_nrf5340_cpunet` to next test too that build hci_rpmsg for iso_receive configuration.

koffes replied:
Missed that one, thanks. Also added to standard hci_rpmsg.

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/board.h:
```diff
@@ -0,0 +1,68 @@
+#define BOARD_REVISION_VALID_MSK_MAX14690_PMIC	\
+	(BOARD_PCA10121_0_6_0_MSK | BOARD_PCA10121_0_7_0_MSK | BOARD_PCA10121_0_7_1_MSK)
+
+#define LED_APP_RGB 0
+#define LED_NET_RGB 1
+#define LED_APP_1_BLUE 2
+#define LED_APP_2_GREEN 3
+#define LED_APP_3_GREEN 4
```
where is this used?

koffes replied:
This is used in an NCS-application LED driver. This is here due to no support for LED grouping. That is, as an RGB led consists of three LEDs, Zephyr does not yet have the required mechanisms to handle that. If must be, we can move these out of the boardfile, but they seem to fit best here.

gmarull replied:
We can't have application-specific code in boards, this is also information that seems to belong to the DT domain.

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,144 @@
+
+	if (!device_is_ready(pmic_iset.port)) {
+		LOG_ERR("GPIO is not ready!");
+		return -ENXIO;
```
`-ENODEV`

koffes replied:
Why do you want -ENODEV here? Do you want to change the other -ENXIOs as well?

nordicjm replied:
Yes

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,144 @@
+		return -ENXIO;
+	}
+
+	ret = gpio_pin_configure_dt(&pmic_iset, GPIO_OUTPUT_LOW);
```
please use `ACTIVE`/`INACTIVE` when using DT APIs.

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,144 @@
+
+	if (!device_is_ready(hw_codec_sel.port)) {
+		LOG_ERR("GPIO is not ready!");
+		return -ENXIO;
+	}
+
+	/* Select on-board HW codec by default */
+	ret = gpio_pin_configure_dt(&hw_codec_sel, GPIO_OUTPUT_LOW);
```
ditto, same for others below.

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts:
```diff
@@ -0,0 +1,132 @@
+		compatible = "gpio-leds";
+		rgb1_red: led_0 {
+			gpios = <&gpio0 7 GPIO_ACTIVE_HIGH>;
+			label = "0 LED_RGB_RED CORE_APP";
```
this has not been fixed. -1 on encoding "structured information" on label, this is not the purpose of such property. If needed downstream, add an overlay there until addressed.

koffes replied:
Roger. When https://github.com/nrfconnect/sdk-nrf/pull/9132 is merged, We can remove some of this.

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts:
```diff
@@ -0,0 +1,132 @@
+		};
+	};
+
+	nrf5340_audio_dk {
+		compatible = "gpio-keys";
+		hw_codec_gpio_in: hw_codec_gpio_in {
+			gpios = <&gpio0 20 GPIO_ACTIVE_LOW>;
+			label = "HW codec GPIO";
+		};
+
```
`gpio-keys` compatible is not meant to be used like this. Please, create a board level compatible to describe such hardware properly. There are a few boards upstream doing this, see e.g.  `boards/arm/96b_wistrio/dts/bindings/qorvo,rfsw8001.yaml`

nordicjm replied:
Unresolved

koffes replied:
Please have a look now. The current solution still requires some polishing. Suggestions are appreciated.

koffes replied:
Right now, all the GPIOs are in the same group. I can create a hw_codec group, pmic group and a general group if that is desired. Then, there would be three files in `dts/bindings`

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_common.dts:
```diff
@@ -0,0 +1,178 @@
+		compatible = "zephyr,sdhc-spi-slot";
+		reg = <0>;
+		status = "okay";
+		mmc {
```
```suggestion
		sdmmc {
```

nordicjm reviewed (commented):
Some unresolved issues to fix

joerchan commented on boards/arm/nrf5340_audio_dk_nrf5340/board.cmake:
```diff
@@ -0,0 +1,20 @@
+board_runner_args(jlink "--device=nrf5340_xxaa_app" "--speed=4000")
+endif()
+
+if(CONFIG_TFM_FLASH_MERGED_BINARY)
+  set_property(TARGET runners_yaml_props_target PROPERTY hex_file "${CMAKE_BINARY_DIR}/tfm_merged.hex")
+endif()
```
So, thingy53 does not have it, we also do not have TF-M support for this board.
This if will never be invoked, and can be removed.

Not a big deal though, just thinking of consistency.
```suggestion
```

joerchan commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,144 @@
+
+	ARG_UNUSED(dev);
+
+	ret = core_config();
```
Apart from the core_config, this code is common for all of the nrf5340 SoC Boards.

Not a comment on your review, but a note for the maintainers of the nRF boards that this should be improved IMO.
This code contains the security configuration of the network core, and it would be nice to have a single implementation of this.

nordicjm reviewed (commented):
non-secure builds fail, for hello_world it does not build with TF-M as the normal nrf5340dk board does, and for `samples/bluetooth/peripheral_hids` it fails to configure:
```
cmake -GNinja -DBOARD=nrf5340_audio_dk_nrf5340_cpuapp_ns ..

warning: SOC_NRF_GPIO_FORWARDER_FOR_NRF5340 (defined at soc/arm/nordic_nrf/nrf53/Kconfig.soc:48) has direct dependencies NRF_SOC_SECURE_SUPPORTED && SOC_NRF5340_CPUAPP && SOC_SERIES_NRF53X && SOC_FAMILY_NRF with value n, but is currently being y-selected by the following symbols:
 - BOARD_ENABLE_CPUNET (defined at boards/arm/nrf5340_audio_dk_nrf5340/Kconfig:23), with value y, direct dependencies BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP || BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP_NS (value: y), and select condition BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP || BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP_NS (value: y)
Parsing /tmp/aa/zephyr/Kconfig
Loaded configuration '/tmp/aa/zephyr/boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_ns_defconfig'
Merged configuration '/tmp/aa/zephyr/samples/bluetooth/peripheral_hids/prj.conf'

error: Aborting due to Kconfig warnings
```

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/dts/bindings/nrf5340_audio_dk.yaml:
```diff
@@ -0,0 +1,55 @@
+      description: Interrupt from HW codec
+
+    hw-codec-reset-out-gpios:
+        type: phandle-array
+        required: true
+        description: Reset HW codec
+
+    hw-codec-sel-out-gpios:
+      type: phandle-array
```
Alignment here changes, use 4 space for main indent then 2 additionally for extra indentation

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/dts/bindings/nrf5340_audio_dk.yaml:
```diff
@@ -0,0 +1,55 @@
+    curr-mon-alert-in-gpios:
+      type: phandle-array
+      required: true
+      description: Board rev readback
```
?

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/dts/bindings/nrf5340_audio_dk.yaml:
```diff
@@ -0,0 +1,55 @@
+
+include: base.yaml
+
+properties:
+    hw-codec-in-gpios:
+      type: phandle-array
+      required: true
+      description: GPIO from HW codec
+
+    hw-codec-irq-in-gpios:
+      type: phandle-array
+      required: true
+      description: Interrupt from HW codec
+
+    hw-codec-reset-out-gpios:
+      type: phandle-array
+      required: true
+      description: Reset HW codec
+
+    hw-codec-sel-out-gpios:
+      type: phandle-array
+      required: true
+      description: HW codec selector
```
this is still not using bindings correctly. Since this is something not needed for upstream, we can add overlays in NCS, so the upstream board doesn't have to deal with hw-codec stuff.

larsgk replied:
Even if the cirrus logic drivers are not upstreamed (yet) - it has value for the upstreamed board definitions to include info on where the HW Codec is hooked up. The board *does* include the HW Codec and it is one of the key values over e.g. nRF5340 DK.

gmarull replied:
Then bindings need to be designed correctly.

koffes replied:
After discussing with @gmarull, this bindings file has been removed.

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/dts/bindings/nrf5340_audio_dk.yaml:
```diff
@@ -0,0 +1,55 @@
+      required: true
+      description: Board rev readback
+
+    curr-mon-alert-in-gpios:
+      type: phandle-array
+      required: true
+      description: Alert from INA231
```
this is wrong: please add entries for INA23X, it is now supported upstream.

nordicjm reviewed (changes requested):
Image is far too large, can you reduce filesize, reduce width/height and use https://imagecompressor.com/

koffes commented:
New compressed image uploaded.

nordicjm commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_cpunet_reset.c:
```diff
@@ -0,0 +1,144 @@
+		return ret;
+	}
+
+	/* Set on-board DSP/HW codec as default */
+	static const struct gpio_dt_spec hw_codec_sel =
+		GPIO_DT_SPEC_GET(DT_NODELABEL(nrf5340_audio_dk), hw_codec_sel_out_gpios);
+
+	if (!device_is_ready(hw_codec_sel.port)) {
+		LOG_ERR("GPIO is not ready!");
+		return -ENODEV;
+	}
+
+	/* Select on-board HW codec by default */
+	ret = gpio_pin_configure_dt(&hw_codec_sel, GPIO_OUTPUT_ACTIVE);
+	if (ret) {
+		return ret;
+	}
```
These seem to go against the board porting guide https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#general-recommendations whilst one may say that's the only reason why someone would use this board (for audio applications) but the same thing applies to the nrf9160dk and having the modem disabled by default, you wouldn't use the nrf9160 unless you wanted a modem, but it is disabled by default to give a minimal interface as per the board porting guide.

cvinayak commented on boards/arm/nrf5340_audio_dk_nrf5340/Kconfig:
```diff
@@ -0,0 +1,58 @@
+	  the board, the application needs to take into consideration, that the
+	  secure firmware image must already have configured GPIO allocation for the
+	  Network MCU.
+	default y if (BT || NRF_802154_SER_HOST)
```
We support 802.15.4? we can tested or is this for future support?

cvinayak reviewed (dismissed):
My minor requests were addressed, other sections seems ok to me.

tejlmand commented on boards/arm/nrf5340_audio_dk_nrf5340/CMakeLists.txt:
```diff
@@ -0,0 +1,15 @@
+
+if ((CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP OR CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP_NS)
+    AND CONFIG_BOARD_ENABLE_CPUNET)
+zephyr_library()
+zephyr_library_sources(nrf5340_audio_dk_cpunet_reset.c)
+
+if (CONFIG_BUILD_WITH_TFM)
+  zephyr_library_include_directories(
+    $<TARGET_PROPERTY:tfm,TFM_BINARY_DIR>/install/interface/include
+  )
+endif()
```
please indent this code:
```suggestion
  zephyr_library()
  zephyr_library_sources(nrf5340_audio_dk_cpunet_reset.c)

  if (CONFIG_BUILD_WITH_TFM)
    zephyr_library_include_directories(
      $<TARGET_PROPERTY:tfm,TFM_BINARY_DIR>/install/interface/include
    )
  endif()
```

tejlmand commented on boards/arm/nrf5340_audio_dk_nrf5340/board.cmake:
```diff
@@ -0,0 +1,16 @@
+endif()
+
+if(CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP OR CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUAPP_NS)
+board_runner_args(jlink "--device=nrf5340_xxaa_app" "--speed=4000")
```
please indent:
```suggestion
  board_runner_args(jlink "--device=nrf5340_xxaa_app" "--speed=4000")
```

tejlmand commented on boards/arm/nrf5340_audio_dk_nrf5340/board.cmake:
```diff
@@ -0,0 +1,16 @@
+endif()
+
+if(CONFIG_BOARD_NRF5340_AUDIO_DK_NRF5340_CPUNET)
+board_runner_args(jlink "--device=nrf5340_xxaa_net" "--speed=4000")
```
please indent:
```suggestion
  board_runner_args(jlink "--device=nrf5340_xxaa_net" "--speed=4000")
```

tejlmand reviewed (changes requested):
some minor nits.

tejlmand reviewed (dismissed):
lgtm

gmarull commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts:
```diff
@@ -0,0 +1,89 @@
+		};
+	};
+
+	nrf5340_audio_dk: nrf5340-audio-dk {
+		compatible = "nrf5340_audio_dk";
+		pmic-iset-out-gpios = <&gpio0 23 GPIO_ACTIVE_HIGH>;
+	};
```
compatible here does not follow the guidelines, following e.g. actinius icarus board (similar case for sim select GPIOs), this could be `nordic-nrf5340-audio-dk`.

joerchan replied:
Don't you mean `nordic,nrf5340-audio-dk`?

gmarull replied:
In the case of actinius a dash was preferred because the sim selector was not a piece of hardware you can buy. Here, this is similar, so probably the name needs to be changed as well. `nordic-pmic-iset` maybe? See https://github.com/zephyrproject-rtos/zephyr/pull/47180#issuecomment-1212208423

carlescufi replied:
So to be clear @gmarull, which change are you suggesting here specifically? Change `nrf5340-audio-dk` to `nordic- nrf5340-audio-dk`? Is this consistent with all of our other DKs?

gmarull replied:
We don't have anything similar on other DKs. The problem with this compatible is that it is not following the naming rules, so it needs to be fixed. The most similar precedent we have is the Actinius Icarus board, where they have a GPIO to indicate SIM selection, and it is configured as part of board init code. At that time, it was agreed to use the `vnd-name` scheme, so here I'd suggest `nordic-pmic-iset`. If the board has a PMIC IC (I thought it was removed according to NCS changes), then the proper solution is to create a regulator driver now that we have infrastructure in place.

anangl commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_common.dts:
```diff
@@ -0,0 +1,223 @@
+	};
+};
+
+zephyr_udc0: &usbd {
+	compatible = "nordic,nrf-usbd";
+	status = "okay";
+};
```
This is a needless copy of: https://github.com/zephyrproject-rtos/zephyr/blob/69fb9991ca48e2fd3b74d3f30bc227e575797920/boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_common.dts#L205-L207.

anangl commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_shared.dts:
```diff
@@ -0,0 +1,89 @@
+/ {
```
Since this file is to be included from other *.dts files, it would be more appropriate to use the .dtsi extension for it. Similarly for nrf5340_audio_dk_nrf5340_shared_sram_planning_conf.dts, nrf5340_audio_dk_nrf5340_cpuapp_common.dts, and nrf5340_audio_dk_nrf5340_cpuapp_partition_conf.dts.

greg-fer commented on boards/arm/nrf5340_audio_dk_nrf5340/doc/index.rst:
```diff
@@ -0,0 +1,109 @@
+     :align: center
+     :alt: nRF5340 DK
+
+More information about the board can be found at the `nRF5340 Audio DK website`_.
```
The hardware for the Audio DK has been published on Infocenter, so we need to rephrase this part:
```suggestion
More information about the board can be found at the `nRF5340 Audio DK website`_. The `Nordic Semiconductor Infocenter`_
contains the processor's information and the datasheet.
```

carlescufi replied:
Fixed

gmarull commented:
last push: added missing defconfig entry

anangl commented on boards/arm/nrf5340_audio_dk_nrf5340/nrf5340_audio_dk_nrf5340_cpuapp_ns_defconfig:
```diff
@@ -0,0 +1,25 @@
+CONFIG_TRUSTED_EXECUTION_NONSECURE=y
+
+CONFIG_GPIO=y
+CONFIG_NFCT_PINS_AS_GPIOS=y
```
This option cannot be enabled here, as non-secure code cannot write UICR, where this configuration needs to be set.

I see that we're missing a related dependency in Kconfig for this option. I'll add it.

gmarull replied:
can you push a fix?

anangl replied:
Sure, I'll do it.

gmarull committed:
drivers: regulator: npm1100: regulator device may be unused

Since nPM1100 may be configured statically in some circuits, BUCK node
may note be defined, so neither API ops or init call are used in the
code. Add __unused attribute to account for such case.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>
